### DISCUSSION TITLE
How to frame url for performing openssl CMP operations using curl apis

### DISCUSSION THREAD

**Discussion by sumanth797** [curl/curl#13970] [Q&A] [answered]
Hi,
I'm framing the URL http://10.33.102.223:80/pkix/ for performing OpenSSL CMP operations and setting the URL using 
**curl_easy_setopt(curl, CURLOPT_URL, URL)**. 

It was working fine.

But when I'm trying to adapt the URL by adding **[]** for both IPV4 and IPV6 it is failing with error,

** URL using bad/illegal format or missing URL** after calling the **curl_easy_perform(curl)** and it is failing for both IPV4 and IPV6

for **IPV4**:
URL = "http://[10.33.102.223]:80/pkix/";
for **IPV6**:
URL = "http://[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:81/pkix/"

How to adapt the change for IPV4 and IPV6? By adding **[]**  it failed for me. 
Can you help me here?
Thanks in Advance.

**Comment by bagder** [comment, marked as answer]
The brackets `[]` should only be there for IPv6 addresses, not for IPv4. Your IPv6 URL works fine to parse here.

**Reply by sumanth797** [reply]
Thanks @bagder, it's working now
